### PR TITLE
xpu seeding PR1

### PR DESCRIPTION
**Pull request by abhilash1910**
xpu basic seeding for mp -1

### CODE REVIEW

File: src/lightning/fabric/utilities/seed.py [hunk 1]
```diff
@@ -104,7 +109,7 @@ def pl_worker_init_function(worker_id: int, rank: Optional[int] = None) -> None:
     random.seed(stdlib_seed)
 
 
-def _collect_rng_states(include_cuda: bool = True) -> Dict[str, Any]:
+def _collect_rng_states(include_cuda: bool = True,include_xpu: bool = True) -> Dict[str, Any]:
```
Comment (jingxu10) [inline src/lightning/fabric/utilities/seed.py]: pls add a space before `include_xpu`

File: src/lightning/fabric/utilities/seed.py [hunk 2]
```diff
@@ -113,6 +118,9 @@ def _collect_rng_states(include_cuda: bool = True) -> Dict[str, Any]:
     }
     if include_cuda:
         states["torch.cuda"] = torch.cuda.get_rng_state_all()
+    if include_xpu:
+        if XPUAccelerator.is_available():
+            states["torch.xpu"] = XPUAccelerator._collect_rng_states()
```
Comment (jingxu10) [inline src/lightning/fabric/utilities/seed.py:123]: better to use `torch.xpu`, rather than `XPUAccelerator`.

File: src/lightning/fabric/utilities/seed.py [hunk 3]
```diff
@@ -123,6 +131,9 @@ def _set_rng_states(rng_state_dict: Dict[str, Any]) -> None:
     # torch.cuda rng_state is only included since v1.8.
     if "torch.cuda" in rng_state_dict:
         torch.cuda.set_rng_state_all(rng_state_dict["torch.cuda"])
+    if "torch.xpu" in rng_state_dict:
+        if XPUAccelerator.is_available():
+            XPUAccelerator._set_rng_states(rng_state_dict)
```
Comment (jingxu10) [inline src/lightning/fabric/utilities/seed.py:136]: better to use `torch.xpu`, rather than `XPUAccelerator`.

File: src/lightning/fabric/utilities/seed.py (hunk 2 above)
Comment (abhilash1910) [reply src/lightning/fabric/utilities/seed.py:123]: Yeah I was thinking the same , but would it not be better to have all abstractions of form "torch.xpu" inside the XPUAccelerator class?

File: src/lightning/fabric/utilities/seed.py (hunk 3 above)
Comment (abhilash1910) [reply src/lightning/fabric/utilities/seed.py:136]: Same as above. if we write torch.xpu here again, then we could have done for all methods. I thought this too hence I wrote in the lightning-xpu class.

**Review by jingxu10 — APPROVED** [review]
